compiler/luci/pass/src/ResolveCustomOpAddPass.test.cpp

```diff
@@ -34,3 +157,73 @@ TEST_F(ResolveCustomOpAddPassTest, name)
   auto const name = _pass.name();
   ASSERT_NE(nullptr, name);
 }
+
+TEST_F(ResolveCustomOpAddPassTest, simple_test_CustomBroadcastTo)
+{
+  _g.init();
+
+  // connect Custom(BroadcastTo) to graph
+  _g.addV2()->inputs(1, _g.custom_broadcastTo_out());
+
+  auto ret = _pass.run(_g.g());
+  EXPECT_EQ(true, ret);
+
+  auto add = dynamic_cast<luci::CircleAdd *>(_g.output(0)->from());
+  EXPECT_NE(nullptr, add);
+
+  auto input_0 = dynamic_cast<luci::CircleNode *>(add->x());
+  EXPECT_NE(nullptr, input_0);
+  EXPECT_EQ(2, input_0->dim(0));
+  EXPECT_EQ(3, input_0->dim(1));
+
+  auto input_1 = dynamic_cast<const luci::CircleConst *>(add->y());
+  EXPECT_NE(nullptr, input_1);
+  EXPECT_EQ(1, input_1->at<loco::DataType::FLOAT32>(0));
+  EXPECT_EQ(2, input_1->at<loco::DataType::FLOAT32>(1));
+  EXPECT_EQ(3, input_1->at<loco::DataType::FLOAT32>(2));
```

Comment:
I can't cat what these checks are doing.
I would help understand with some explanation comments.

Reply:
> // check if inputs are set well after changing from Custom(AddV2) to Add.

I mean what are the numbers like 0 and 1 in
```
  EXPECT_EQ(2, input_0->dim(0));
  EXPECT_EQ(3, input_0->dim(1));
```
and 0, 1, 2 in
```
  EXPECT_EQ(1, input_1->at<loco::DataType::FLOAT32>(0));
  EXPECT_EQ(2, input_1->at<loco::DataType::FLOAT32>(1));
  EXPECT_EQ(3, input_1->at<loco::DataType::FLOAT32>(2));
```
?

Reply:
> set well

word `well` is ambiguous . plz be more specific.

Reply:
@seanshpark 

I changed `input_0` and `input_1` to `input_x` and `input_y` and the comment to make it easier to understand as follows.

```
  // check input_x and input_y to see if Custom(AddV2) is resolved.
  auto input_x = dynamic_cast<luci::CircleNode *>(add->x());
  EXPECT_NE(nullptr, input_x);
  EXPECT_EQ(2, input_x->dim(0));
  EXPECT_EQ(3, input_x->dim(1));

  auto input_y = dynamic_cast<const luci::CircleConst *>(add->y());
  EXPECT_NE(nullptr, input_y);
  EXPECT_EQ(1, input_y->at<loco::DataType::FLOAT32>(0));
  EXPECT_EQ(2, input_y->at<loco::DataType::FLOAT32>(1));
  EXPECT_EQ(3, input_y->at<loco::DataType::FLOAT32>(2));
```

And FYI, Because `CircleInput` and input of `BroadcastTo` change to `x` and `y` if it is resolved from `Custom(AddV2)` to `Add`, so check the value of each dim here.
Would it be better to add more comments to explain this? :)

Reply:
1/ what's the difference of changing `input_0` to `intput_x` ?

2/ I asked about the numbers, not the variables.
to add more, why are the values `2, 3` and `1, 2, 3` compared?
what are the meanings of these number?

3/ what are the purpose of these tests?

Reply:
@seanshpark 

> 1/ what's the difference of changing input_0 to intput_x ?

There is no meaningful difference. 😅
but, It's just... To understand the test easily, the names of the variables are matched to the options `x` and `y` of `Add`.

> 2/ I asked about the numbers, not the variables.
to add more, why are the values 2, 3 and 1, 2, 3 compared?
what are the meanings of these number?

the value 2, 3 are the **shape** of `CircleInput`, 
and values 1, 2, 3 are the **values** of `BroadcastTo-input()`.
--->
```
    // CircleConst (BroadcastTo-input)
    auto input = g->nodes()->create<luci::CircleConst>();
    input->dtype(loco::DataType::FLOAT32);
    input->shape({1, 3});
    input->size<loco::DataType::FLOAT32>(3);
    input->at<loco::DataType::FLOAT32>(0) = 1;
    input->at<loco::DataType::FLOAT32>(1) = 2;
    input->at<loco::DataType::FLOAT32>(2) = 3;
```


> 3/ what are the purpose of these tests?
if `Custom(AddV2)` is resolved to `Add` through pass, because `CircleInput` and `BroadcastTo->input()` are connected to `Add`'s x and y, so I test to confirm this.

Reply:
> the value 2, 3 are the shape of CircleInput,

really?

Reply:
@seanshpark 

> plz add some comments about this and use DEFINES not just 1, 2, 3 so that readers can understand what is check and why it is checked.

Okay, I will add comments as you said.
Thank you for your advice from the user's point of view all the time. :)


> really? why do we check this?
what's the point of pass?


Well... I'm sorry that I made a mistake of setting the shape in `CircleInput` instead of the values of input. 😂

Before making the test codes, I referred to the existing [ResolveCustomOpSplitVPass.test.cpp](https://github.com/Samsung/ONE/blob/356901d7a13c270252919834b581096e2cf35663/compiler/luci/pass/src/ResolveCustomOpSplitVPass.test.cpp#L113), [ResolveCustomOpBatchMatMulPass.test.cpp](https://github.com/Samsung/ONE/blob/356901d7a13c270252919834b581096e2cf35663/compiler/luci/pass/src/ResolveCustomOpBatchMatMulPass.test.cpp#L101), and i misunderstood that only the shape was set in `CircleInput`.
so, it seems that the test needs to be modified to **check the values of CircleInput** rather than to check the shape of CircleInput.

And this is the other part, 
As I thought about it while answering your question, this pass only contains the function of converting `Custom(AddV2)` to `Add`.
Therefore, the test only needs to check whether `x` and `y` of `Add` are set without checking the values. The code for this is as follow, Could I change it like this?

```
TEST_F(ResolveCustomOpAddPassTest, simple_test_CustomBroadcastTo)
{
  _g.init(true);

  auto ret = _pass.run(_g.g());
  EXPECT_EQ(true, ret);

  auto add = dynamic_cast<luci::CircleAdd *>(_g.output(0)->from());
  EXPECT_NE(nullptr, add);

  <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
  // check input_x and input_y to see if Custom(AddV2) is resolved.
  // if the pass is run, CircleInput will be connected to x of add and BroadcastTo->input() will be connected to y of add.
  auto input_x = dynamic_cast<luci::CircleNode *>(add->x());
  EXPECT_NE(nullptr, input_x);

  auto input_y = dynamic_cast<const luci::CircleConst *>(add->y());
  EXPECT_NE(nullptr, input_y);
 <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
}
```

Reply:
@seanshpark 

> plz add some comments about this and use DEFINES not just 1, 2, 3 so that readers can understand what is check and why it is checked.

Okay, I will add comments as follows. Thank you a lot :)
```
  // check if inputs are set after changing from Custom(AddV2) to Add.
  // if the pass runs, x and y of Custom(AddV2) will be connected to x and y of Add.
```

> really? why do we check this?
what's the point of pass?

well... When the pass is executed, x and y of `Custom(AddV2)` are connected to x and y of `Add`. 
Then, since the shape and values of x and y are maintained even after they are connected, 
I think this should be checked.

Reply:
>  and use DEFINES not just 1, 2, 3 so that